lib/rucio/core/transfer.py

```diff
@@ -702,6 +781,271 @@ def __rewrite_dest_url(dest_url, dest_sign_url, dest_scheme):
     return dest_url
 
 
+def __prepare_source_dest_config(ctx, request_id, src_rse_id, dest_rse_id, source_scheme, dest_scheme, dict_attributes, activity,
+                                 scope, name, file_path, retry_count, reqs_only_tape_source, reqs_no_source, bring_online_local):
+
+    # Get source protocol
+    source_protocol = ctx.protocol(src_rse_id, source_scheme, 'read')
```

Comment:
read -> third_party_copy 
Was this read before in the old code?

Reply:
it was . it's just that ` ctx.protocol(src_rse_id, source_scheme, 'read')` doesn't correctly use this parameter . This issue was mentioned by @cserf in the previous refactoring. However, the previous refactoring tried to keep the same behavior as before. This time it's the moment to fix it. I'll make the change.

Reply:
done